Sources/Storage/NetworkDriver.swift

```diff
@@ -65,13 +91,20 @@ public final class S3Driver: NetworkDriver {
             print("Please check `template` in `storage.json`.")
             throw Error.pathMissingForwardSlash
         }
-
+        
         return try s3.upload(
             bytes: Data(bytes),
             path: path,
-            access: .publicRead,
+            contentType: mime,
+            access: access,
             on: container
-        ).map { _ in
+        ).map { res in
+            guard
+                res.http.status.code == 200
```

Comment:
```suggestion
                res.http.status == .ok
```

Would that work? Then we can rely on Vapor's definition of ok